src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -166,34 +181,39 @@ public async Task<SymbolInformation[]> GetWorkspaceSymbolsAsync(JToken input, Ca
             return await _protocol.GetWorkspaceSymbolsAsync(_workspace.CurrentSolution, workspaceSymbolParams, _clientCapabilities, cancellationToken).ConfigureAwait(false);
         }
 
+        private void DiagnosticService_DiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
```

Comment:
Since this is still the one triggered by the event handler, this should still be async void and wrapped in the try catch instead of in PublishDiagnosticsAsync.